src/reducer.ts

```diff
@@ -108,50 +139,120 @@ type ReducerCreator = {
     actionCreator: UnknownActionCreator[],
     reducer: Reducer<State, VoidPayload>
   ): RegisteredReducer<State, VoidPayload>;
+
+  <State, Payload>(
+    source$: Observable<Payload>,
+    reducer: Reducer<State, Payload>
+  ): RegisteredReducer<State, Payload>;
 };
 
 export const reducer: ReducerCreator = <State>(
```

Comment:
Great idea to overload this and just treat this as two types of reducers with either an action or a stream payload :)